### PR TITLE
Require semantic version for model parameter version string

### PR DESCRIPTION
**Pull request by GernotMaier**
Model versions have changed to semantic versioning with this [model_parameters PR](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters/-/merge_requests/50).

Add here:

- a test to validate_data that the version string follows semantic versioning.
- change of "2024-02-01" to prod6 and "2020-06-28" to prod5
- run `validate_camera_efficiency_ssts` test for prod6 only (as Derived-DB is not updated with the new versioning; see issue #1070)

### CODE REVIEW

File: simtools/data_model/validate_data.py [hunk 1]
```diff
@@ -708,3 +710,25 @@ def _prepare_model_parameter(self):
...
+
+        Raises
+        ------
+        ValueError
+            if version string does not follow semantic versioning
+
+        """
+        if version is None:
+            return
+        semver_regex = r"^\d+\.\d+\.\d+(-[0-9A-Za-z.-]+)?(\+[0-9A-Za-z.-]+)?$"
```
Comment (tobiaskleiner) [inline simtools/data_model/validate_data.py:731]: Not perfectly familiar with regex. Does this here allow also number.number.number-(additional expression)?

File: tests/unit_tests/data_model/test_validate_data.py [hunk 2]
```diff
@@ -695,3 +696,41 @@ def test_prepare_model_parameter():
...
+
+
+def test_check_version_string(caplog):
+    data_validator = validate_data.DataValidator()
+
+    valid_versions = [
+        "1.0.0",
+        "0.1.0",
+        "2.3.4",
+        "1.0.0-alpha",
```
Comment (tobiaskleiner) [inline tests/unit_tests/data_model/test_validate_data.py:708]: Related to above question. So we really want to allow this kind of version structure?

Comment (GernotMaier) [reply tests/unit_tests/data_model/test_validate_data.py:708]: This is how semantic versioning is defined, see https://semver.org/

**Review by tobiaskleiner — COMMENTED** [review]
I put this comment here that this would also accept pre release versions.

File: tests/unit_tests/data_model/test_validate_data.py (hunk 2 above)
Comment (tobiaskleiner) [reply tests/unit_tests/data_model/test_validate_data.py:708]: Ok all good then.

**Review by tobiaskleiner — APPROVED** [review]

File: simtools/data_model/validate_data.py (hunk 1 above)
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:731]: Yes - but see also #1115

**Pull request merged**